Description: Use system libcrypto++ if available
 Rather than always use the included libcrypto++ zipped source, check
 for and use the system's libcrypto++.
Author: Jeff Licquia <licquia@debian.org>
Forwarded: no
Last-Update: 2013-06-01
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
Index: synergy-1.8.7/CMakeLists.txt
===================================================================
--- synergy-1.8.7.orig/CMakeLists.txt	2017-01-19 18:01:47.803556127 +0100
+++ synergy-1.8.7/CMakeLists.txt	2017-01-19 18:01:47.795556219 +0100
@@ -91,6 +91,7 @@
 	include(CheckIncludeFileCXX)
 	include(CheckSymbolExists)
 	include(CheckCSourceCompiles)
+	include(FindPkgConfig)
 
 	check_include_file_cxx(istream HAVE_ISTREAM)
 	check_include_file_cxx(ostream HAVE_OSTREAM)
@@ -234,6 +235,8 @@
 		check_library_exists("Xi" XISelectEvents "" HAVE_Xi)
 		check_library_exists("Xrandr" XRRQueryExtension "" HAVE_Xrandr)
 
+		pkg_search_module(CRYPTOPP libcrypto++)
+
 		if (HAVE_ICE)
 
 			# Assume we have SM if we have ICE.
@@ -277,6 +280,10 @@
 			list(APPEND libs Xi)
 		endif()
 
+		if (HAVE_CRYPTOPP)
+			list(APPEND libs CRYPTOPP_LIBRARIES)
+		endif()
+
 	endif()
 
 	# For config.h, set some static values; it may be a good idea to make
